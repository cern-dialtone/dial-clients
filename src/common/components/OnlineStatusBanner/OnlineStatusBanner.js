import React, { useEffect, useCallback } from 'react';
import { Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import { logMessage } from 'common/utils/logs';
import styles from './OnlineStatusBanner.module.css';

function OnlineStatusBanner({ onlineStatus, setOnlineStatus }) {
  logMessage('Loading OnlineStatusBanner');

  const setAppOnlineStatus = useCallback(() => {
    setOnlineStatus(navigator.onLine);
  }, [setOnlineStatus]);

  useEffect(() => {
    setAppOnlineStatus();
  }, [setAppOnlineStatus]);

  useEffect(() => {
    logMessage('Calling useEffect');
    window.addEventListener('online', setAppOnlineStatus);
    window.addEventListener('offline', setAppOnlineStatus);
    return () => {
      window.removeEventListener('online', setAppOnlineStatus);
      window.removeEventListener('offline', setAppOnlineStatus);
    };
  }, [setAppOnlineStatus]);

  if (onlineStatus) {
    return null;
  }
  return (
    <div className={`padded-item ${styles.callForwardingMessage}`}>
      <Icon name="warning sign" /> You are currently offline. Please, check your
      network connection
    </div>
  );
}

OnlineStatusBanner.propTypes = {
  onlineStatus: PropTypes.bool.isRequired,
  setOnlineStatus: PropTypes.func.isRequired
};

export default OnlineStatusBanner;
