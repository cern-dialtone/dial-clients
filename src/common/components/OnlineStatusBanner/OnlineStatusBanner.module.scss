.callForwardingMessage {
  background: #db2828;
  min-height: 30px;
  position: fixed;
  top: 0;
  display: flex;
  width: 100%;
  box-sizing: border-box;
  z-index: 9999;
  cursor: pointer;
  color: #fff;
  text-align: center;
}
