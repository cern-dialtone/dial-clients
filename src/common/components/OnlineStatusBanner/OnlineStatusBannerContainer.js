import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { setOnlineStatus } from 'settings/actions/settings';
import OnlineStatusBanner from './OnlineStatusBanner';

function mapStateToProps({ settings }) {
  return {
    onlineStatus: settings.settings.onlineStatus
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      setOnlineStatus
    },
    dispatch
  );
}

export const ConnectedOnlineStatusBanner = connect(
  mapStateToProps,
  mapDispatchToProps
)(OnlineStatusBanner);

export default ConnectedOnlineStatusBanner;
