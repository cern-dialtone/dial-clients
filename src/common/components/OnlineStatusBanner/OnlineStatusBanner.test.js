import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import OnlineStatusBanner from './OnlineStatusBanner';

global.remote = {
  require: jest.fn()
};

describe('OnlineStatusBanner tests', () => {
  const setOnlineStatus = jest.fn();

  it('renders component with onlineStatus=true', async () => {
    const { container } = render(
      <OnlineStatusBanner onlineStatus setOnlineStatus={setOnlineStatus} />
    );
    expect(container.firstChild).toBeNull();
  });

  it('renders null without any errors', async () => {
    const { asFragment, getByText } = render(
      <OnlineStatusBanner
        onlineStatus={false}
        setOnlineStatus={setOnlineStatus}
      />
    );
    await waitForElement(() =>
      getByText(
        'You are currently offline. Please, check your network connection'
      )
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
