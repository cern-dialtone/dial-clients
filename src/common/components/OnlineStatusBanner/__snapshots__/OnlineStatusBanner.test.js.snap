// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineStatusBanner tests renders null without any errors 1`] = `
<DocumentFragment>
  <div
    class="padded-item callForwardingMessage"
  >
    <i
      aria-hidden="true"
      class="warning sign icon"
    />
     You are currently offline. Please, check your network connection
  </div>
</DocumentFragment>
`;
